robottelo/hosts.py

```diff
@@ -1753,3 +1763,169 @@ def register_host_custom_repo(self, module_org, rhel_contenthost, repo_urls):
         )
         # refresh repository metadata on the host
         rhel_contenthost.execute('subscription-manager repos --list')
+
+
+class SSOHost(Satellite):
+    def __init__(self, **kwargs):
+        kwargs['hostname'] = kwargs.get('hostname', settings.rhsso.host_name)
+        super().__init__(**kwargs)
+
+    def get_rhsso_client_id(self, sat_obj):
+        """getter method for fetching the client id and can be used other functions"""
+        client_name = f'{sat_obj.hostname}-foreman-openidc'
```

Comment:
I dont see this variable is used at all, please remove it !

Reply:
Also sat_obect is also not used anywhere. TBH I am not sure why we need sat_obj as a parameter. Isnt the SSOHost inheriting the Satellite class could be used as `self.hostname` to identify that satellite object.

Reply:
I personally think SSOHost should not inherit the Satellite Class, It would rather have the Satellite object as part of the composition rather than inheriting.